Dockerfile

```diff
@@ -1,27 +1,89 @@
-FROM golang:1.13.4 as git-compiler-helper
-ADD ./git-compiler-helper /root/git-compiler-helper
-WORKDIR /root/git-compiler-helper
-RUN go get -d ./...
-RUN go build -o /root/git-latest-url-finder.run .
-
 FROM ubuntu:latest as git-compiler
-ARG DEBIAN_FRONTEND=noninteractive
-RUN apt-get update
-RUN apt-get install build-essential autoconf dh-autoreconf -y
-RUN apt-get install libcurl4-openssl-dev tcl-dev gettext \
-  asciidoc libexpat1-dev libz-dev -y
-RUN apt-get install curl -y
-RUN mkdir -p /root/src
+# Optional curl options like --insecure in case you're behind a corporate proxy.
+ARG CURL_OPTIONS=''
+# no interactions with apt possible
+ENV DEBIAN_FRONTEND=noninteractive
+# update apt
+RUN apt-get update -qq -y;
+# Install stuff for git itself
+RUN apt-get install -qq -y --no-install-recommends \
+    zlib1g \
+    libcurl4 \
+    perl \
+    libpcre3 \
+    libexpat1 \
+    liberror-perl \
+    libc6;
+# Install meta-things for building on the end-user's machine
+RUN apt-get install -qq -y --no-install-recommends \
+    build-essential \
+    autoconf \
+    dh-autoreconf \
+    # @see https://help.ubuntu.com/community/CheckInstall
+    checkinstall \
+    curl \
+    # @see https://stedolan.github.io/jq/tutorial/
+    jq;
+# Install dev stuff for git
+RUN apt-get install -qq -y --no-install-recommends \
+    libcurl4-openssl-dev \
+    tcl-dev \
+    gettext \
+    asciidoc \
+    libexpat1-dev \
+    libz-dev \
+    libssl-dev;
+# prepare source dir
+RUN mkdir -p /root/src;
 WORKDIR /root/src
-COPY --from=git-compiler-helper \
-  /root/git-latest-url-finder.run \
-  /root/git-latest-url-finder.run
-RUN /root/git-latest-url-finder.run > git-tar-url.txt && \
-  echo "$(cat git-tar-url.txt)" && \
-  curl -L --retry 5 --url "$(cat git-tar-url.txt)" --output "git-source.tar.gz" && \
-  tar -xf "git-source.tar.gz" --strip 1
-RUN make configure
-RUN CFLAGS="-static" ./configure --with-openssl
-RUN make
-RUN make install
-ENTRYPOINT [ "git" ]
\ No newline at end of file
+# download & extract source tar
+RUN curl ${CURL_OPTIONS} \
+        --location \
+        --fail \
+        # assuming the top-most tag is the most recent version
+        $(curl ${CURL_OPTIONS} --silent --fail 'https://api.github.com/repos/git/git/tags' | jq '.[0].tarball_url' | tr -d '"') \
```

Comment:
Oh SWEET! I thought you had to have an API key to talk to Github's API! I didn't know you could do this anonymously :D